cloud/openstack/os_server_group.py

```diff
@@ -0,0 +1,163 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+module: os_server_group
+short_description: Manage OpenStack Server Groups
+extends_documentation_fragment: openstack
+version_added: "2.2"
+author: "Tim Laszlo (@timl)"
+description:
+    - Manage OpenStack server groups. Groups can be created or
+      deleted using this module. Updating existing groups is
+      not supported by openstack.
+options:
+   name:
+     description:
+        - Name of the server group
+     required: true
+   policies:
+     description:
+        - List of policies for the server group.
+     required: false
+     choices: [affinity, anti-affinity]
+     default: None
+   state:
+     description:
+       - Should the resource be present or absent.
+     choices: [present, absent]
+     default: present
+requirements:
+    - "python >= 2.6"
+    - "shade"
+'''
+
+EXAMPLES = '''
+# Create a server group
+- os_server_group:
+    state: present
+    name: friends
+    policies:
+      - affinity
+
+# Delete a server group
+- os_server_group:
+    state: absent
+    name: friends
+'''
+
+RETURN = '''
+server_group:
+    description: Dictionary describing the server group.
+    returned: On success when I(state) is 'present'
+    type: dictionary
+    contains:
+        id:
+            description: Security Group ID
+            returned: success
+            type: string
+            sample: "cb4f45b4-08b6-4ea5-b4c7-6d39ec402e03"
+        name:
+            description: Security group
+            returned: success
+            type: string
+            sample: "security_group"
+        policies:
+            description: Policies applied to this group
+            returned: success
+            type: list
+            sample: "[affinity]"
+        members:
+            description: Hosts in this group
+            returned: success
+            type: list
+            sample: "[hostX]"
+        metadata:
+            description: Metadata for this group
+            returned: success
+            type: complex
+'''
+
+
+def _system_state_change(state, server_group):
+    if state == 'present' and not server_group:
+        return True
+    if state == 'absent' and server_group:
+        return True
+    return False
+
+
+def main():
+
+    argument_spec = openstack_full_argument_spec(
+        name=dict(required=True),
+        policies=dict(required=False, default=None),
```

Comment:
You have a good point.  Policies is defined as a list to match the underlying OpenStack api.  I modeled the option after that anticipating more options may be available in the future.  Since AnsibleModule's choices option matches the entire list, it would be a poor choice to use directly.

Would it be better to hint which policies may be available in the docs instead of using "choices"?  Alternatively, I could match each list item in my module and validate individually.

Thank you for the review and feedback.